### PR TITLE
Add ability to set compression level or disable compression

### PR DESCRIPTION
**Pull request by pvint**
Based on the discussion in #32 I've added the ability to set compression level (1-9) or disable compression:

image

Summary:
Set the compression level from 1-9, or set to 0 to disable compression and not run 7Zip at all. Tested and working for both backup and restore.

### CODE REVIEW

**Comment by mrrfv** [comment]
Thank you so much! It's getting late where I live, so I'll review your PR tomorrow.

File: backup.sh [hunk 1]
```diff
@@ -280,39 +309,45 @@ then
...
   if [ ! -v archive_path ]; then
-    text_input "Please provide the location of the backup archive to restore (drag-n-drop):" archive_path
+    text_input "Please provide the location of the backup directory/archive to restore (drag-n-drop):" archive_path
   fi
 
-  if [ ! -f "$archive_path" ]; then
+  if [ ! -f "$archive_path" -a "${archive_path##*.}" = "7z" ]; then
       cecho "The specified backup location doesn't exist or isn't a file."
       exit 1
+  # If we're restoring from uncompressed backup, make sure at lease the Apps directory exists
```
Comment (mrrfv) [inline backup.sh:318]:
```suggestion
  # If we're restoring from uncompressed backup, make sure at least the Apps directory exists
```

kinda nitpicking here

File: backup.sh [hunk 2]
```diff
@@ -255,15 +282,17 @@ then
...
-  retry 5 7z a -p$archive_password -mhe=on -mx=9 -bb3 -sdel $backup_archive backup-tmp/*
+  if [ "$compression_level" != 0 ]; then
+    cecho "Compressing & encrypting data - this will take a while."
+    # -p: encrypt backup
+    # -mhe=on: encrypt headers (metadata)
+    # -mx=9: ultra compression
+    # -bb3: verbose logging
+    # -sdel: delete files after compression
+    # The undefined variable is set by the user
+    declare backup_archive="$archive_path/linux-android-backup-$(date +%m-%d-%Y-%H-%M-%S).7z"
```
Comment (mrrfv) [inline backup.sh:293]: in order to ensure compatibility with hooks, `backup_archive` should be defined regardless of whether the user's using compression or not.

https://github.com/mrrfv/linux-android-backup/blob/0fa07936f958801d027d29758a08235d8f9c6acd/README.md?plain=1#L81

**Review by mrrfv — CHANGES_REQUESTED** [review]
I reviewed your changes a few days ago, and didn't realize that I have to submit the review for you to see my feedback :p

**Comment by pvint** [comment]
The changes you suggest make sense and look good - I'll try to do them tomorrow and re-test.

Paul

**Comment by kaktus40** [comment]
Hi, I'm very interested in disabled compression...

**Comment by mrrfv** [comment]
@kaktus40 You can use the `discouraged_disable_archive` environment variable if you really need to disable compression and encryption, see [here](https://github.com/mrrfv/open-android-backup#automationunattended-backups) for details.

**Comment by 3eggert** [comment]
perhaps my pr:  

make compression level adjustable #105 

can help since this one seems abandoned.

**Pull request closed without merging**